### DISCUSSION TITLE
How ParquetSharp handle legacy parquet files

### DISCUSSION THREAD

**Discussion by Mydurian** [G-Research/ParquetSharp#429] [General]
Hi Team,
When referring the Apache official doc for complex logical type.. I see that the structure definition is somewhere different. Could you help elaborate a little bit more on how to handle some backward-compatibility in ParquetSharp? Thanks in advance.
<img width="882" alt="image" src="https://github.com/G-Research/ParquetSharp/assets/48521631/6132d709-a399-47a7-b825-4203ca47ab1e">

https://github.com/apache/parquet-format/blob/master/LogicalTypes.md

**Comment by adamreeve** [comment]
Hi @Mydurian, ParquetSharp doesn't depend on the names of schema nodes matching the specification, following the backward-compatibility rules, but we currently only support lists that are represented with the three level structure and don't handle data that is missing the inner element layer.

I don't think there's a simple way to work around this, but we'd be open to accepting a pull request to add backwards-compatibility support for reading files that don't use the standard schema structure.
